include/zephyr/drivers/charger.h

```diff
@@ -169,6 +171,8 @@ union charger_propval {
 	enum charger_charge_type charge_type;
 	/** CHARGER_PROP_HEALTH */
 	enum charger_health health;
+	/** CHARGER_PROP_CONSTANT_CHARGE_CURRENT */
+	uint32_t const_charge_curr;
```

Comment:
I think we are leaving an opportunity on the table here: one of the problem I bumped into all the time with the sensor subsystem is that driver are messing up the scaling in the values. Here since we have the enums we can put the unit in the name.

How about `const_charge_curr_ua` and so on...? That would put the unit where it's needed in both the driver and application code rather than in the generic header and documentation that no reviewer remembers to check anyway.

@teburd @MaureenHelm @yperess @aaronemassey @gmarull @aasinclair

opinion?

Reply:
I am okay with this. ack.

Reply:
I think that's a good idea! I'd almost always rather be explicit when possible.

Reply:
done